src/VisualQueryBuilder/components/OperationEditorBody.tsx

```diff
@@ -0,0 +1,288 @@
+import { getOperationParamEditor, getOperationParamId } from './OperationParamEditor';
+import { Stack } from '../../QueryEditor/Stack';
+import { v4 } from 'uuid';
+
+type Props = {
+    provided: DraggableProvided;
+    isConflicting: boolean,
+    index: number,
+    operation: QueryBuilderOperation,
+    definition: QueryBuilderOperationDefinition,
+    queryModeller: VisualQueryModeller;
+    query: VisualQuery;
+    onChange: (index: number, update: QueryBuilderOperation) => void;
+    onRemove: (index: number) => void;
+    onRunQuery: () => void;
+    datasource: DataSourceApi;
+    flash?: boolean,
+    highlight?: boolean,
+    timeRange?: TimeRange;
+  }
+  
+export function OperationEditorBody({ provided, flash, isConflicting, highlight, index, queryModeller, onChange, onRemove, operation, definition, query, timeRange, onRunQuery, datasource }: Props) {
+  const theme = useTheme2();
+  const styles = getStyles(theme, isConflicting);
+  const shouldFlash = useFlash(flash);
+  const { current: id } = useRef(v4());
+
+  const onParamValueChanged = (paramIdx: number, value: QueryBuilderOperationParamValue) => {
+    const update: QueryBuilderOperation = { ...operation, params: [...operation.params] };
+    update.params[paramIdx] = value;
+    callParamChangedThenOnChange(definition, update, index, paramIdx, onChange);
+  };
+
+  const onAddRestParam = () => {
+    const update: QueryBuilderOperation = { ...operation, params: [...operation.params, ''] };
+    callParamChangedThenOnChange(definition, update, index, operation.params.length, onChange);
+  };
+
+  const onRemoveRestParam = (paramIdx: number) => {
+    const update: QueryBuilderOperation = {
+      ...operation,
+      params: [...operation.params.slice(0, paramIdx), ...operation.params.slice(paramIdx + 1)],
+    };
+    callParamChangedThenOnChange(definition, update, index, paramIdx, onChange);
+  };
+
+  // Handle adding button for rest params
+  let restParam: React.ReactNode | undefined;
+  if (definition.params.length > 0) {
+    const lastParamDef = definition.params[definition.params.length - 1];
+    if (lastParamDef.restParam) {
+      restParam = renderAddRestParamButton(lastParamDef, onAddRestParam, index, operation.params.length, styles);
+    }
+  }
+
+    return (
+      <div
+      className={cx(
+        styles.card,
+        (shouldFlash || highlight) && styles.cardHighlight,
+        isConflicting && styles.cardError
+      )}
+      ref={provided.innerRef}
+      {...provided.draggableProps}
+      data-testid={`operations.${index}.wrapper`}
+    >
+      <OperationHeader
+        operation={operation}
+        dragHandleProps={provided.dragHandleProps}
+        definition={definition}
+        index={index}
+        onChange={onChange}
+        onRemove={onRemove}
+        queryModeller={queryModeller}
+      />
+      <div className={styles.body}>
+      {
```

Comment:
This looks weirdly styled, maybe it needs to pass through prettier.

Reply:
Fixed in https://github.com/grafana/grafana-experimental/pull/112/commits/eadd757e3bf0f18bc6e6418139927eeb29a1c14a